src/main/java/com/lothrazar/storagenetwork/item/ItemCollector.java

```diff
@@ -60,12 +59,16 @@ public void onEntityItemPickupEvent(EntityItemPickupEvent event) {
         BlockEntity tile = serverTargetWorld.getBlockEntity(dp.getBlockPos());
         if (tile instanceof TileMain) {
           TileMain network = (TileMain) tile;
-          //
-          int countUnmoved = network.insertStack(item, false);
+
+          // Create a new reference to the stack, try to insert that into the
+          // network, then change the original stack size so the player picks up
+          // only what remains, if anything.
+          int countUnmoved = network.insertStack(item.copy(), false);
+          item.setCount(countUnmoved);
```

Comment:
Minecraft already handles picking up an empty stack as we need it to; but the player should only receive the _remainder_ after the inventory network takes as much as it can, not the entire stack.

If the network can hold none of the copied stack, it should be cleaned up by GC since it does not actually exist in the 3D world. The player should pick up the stack as usual since the remainder _is_ the entire stack.